boot/flags.go

```diff
@@ -269,6 +269,8 @@ func setNextBootFlags(dev snap.Device, rootDir string, flags []string) error {
 		return errNotUC20
 	}
 
+	// XXX take the modeenv lock?
```

Comment:
Why it cannot be done yet here?

Reply:
because I'm trying to do locking at the entry points into this package, but actually this seems to be an unused unexported helper, so it's a bit unclear how to proceed at this point in time, a higher level function that actually uses it would do the locking usually